core/src/main/java/nl/nn/adapterframework/filesystem/FileSystemBase.java

```diff
@@ -36,6 +39,7 @@
 	private int maxNumberOfMessagesToList=-1;
 	
 	private boolean open;
+	private @Getter @Setter String charset = StreamUtil.DEFAULT_INPUT_STREAM_ENCODING;
```

Comment:
It is not the FileSystem  itself that should get the charset, only the actor.
Please change the interface IBasicFileSystem to include a charset argument in readFile()